src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -760,7 +762,7 @@ def update_file_service_properties(cmd, instance, enable_delete_retention=None,
         instance.protocol_settings.smb.kerberos_ticket_encryption = kerberos_ticket_encryption
     if channel_encryption is not None:
         instance.protocol_settings.smb.channel_encryption = channel_encryption
-    if any(instance.protocol_settings.smb.__dict__.values()):
+    if instance.protocol_settings and instance.protocol_settings.smb and any(instance.protocol_settings.smb.__dict__.values()):
```

Comment:
I don't think this PR is a good idea.

If we implement this check, we should implement checks for all object properties, like this `protocolSettings` or `shareDeleteRetentionPolicy`, just in case they are suddenly missing from the API's response.

```
> az storage account file-service-properties show -n clitest23zhehg2ug7pzcmmt
{
  "cors": {
    "corsRules": []
  },
  "id": "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/clitest47xrljy3nijo3qd5vzsdilpqy5gmhc6vhrxdt4iznh6uaopskftgp4scam2w7drpot4l/providers/Microsoft.Storage/storageAccounts/clitest23zhehg2ug7pzcmmt/fileServices/default",
  "name": "default",
  "protocolSettings": null,
  "resourceGroup": "clitest47xrljy3nijo3qd5vzsdilpqy5gmhc6vhrxdt4iznh6uaopskftgp4scam2w7drpot4l",
  "shareDeleteRetentionPolicy": {
    "allowPermanentDelete": null,
    "days": 0,
    "enabled": false
  },
  "sku": {
    "name": "Standard_LRS",
    "tier": "Standard"
  },
  "type": "Microsoft.Storage/storageAccounts/fileServices"
}
```

Also, after API fixes this, the check for `protocolSettings` will be rendered useless.